x-pack/plugins/enterprise_search/server/lib/check_access.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { KibanaRequest, Logger } from 'src/core/server';
+import { SecurityPluginSetup } from '../../../security/server';
+import { ConfigType } from '../';
+
+import { callEnterpriseSearchConfigAPI } from './enterprise_search_config_api';
+
+interface ICheckAccess {
+  request: KibanaRequest;
+  security?: SecurityPluginSetup;
+  config: ConfigType;
+  log: Logger;
+}
+export interface IAccess {
+  hasAppSearchAccess: boolean;
+  hasWorkplaceSearchAccess: boolean;
+}
+
+const ALLOW_ALL_PLUGINS = {
+  hasAppSearchAccess: true,
+  hasWorkplaceSearchAccess: true,
+};
+const DENY_ALL_PLUGINS = {
+  hasAppSearchAccess: false,
+  hasWorkplaceSearchAccess: false,
+};
+
+/**
+ * Determines whether the user has access to our Enterprise Search products
+ * via HTTP call. If not, we hide the corresponding plugin links from the
+ * nav and catalogue in `plugin.ts`, which disables plugin access
+ */
+export const checkAccess = async ({
+  config,
+  security,
+  request,
+  log,
+}: ICheckAccess): Promise<IAccess> => {
+  // If security has been disabled, always show the plugin
+  if (!security?.authz?.mode.useRbacForRequest(request)) {
```

Comment:
NIT: `security` plugin dep is optional, `authz` on the enabled plugin is not. -> `security?.authz.mode.useRbacForRequest(request)`